sw/device/lib/crypto/drivers/kmac.h

```diff
@@ -0,0 +1,86 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+#ifndef OPENTITAN_SW_DEVICE_LIB_CRYPTO_DRIVERS_KMAC_H_
+#define OPENTITAN_SW_DEVICE_LIB_CRYPTO_DRIVERS_KMAC_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/macros.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum kmac_operation {
+  kSHA3_224,
+  kSHA3_256,
+  kSHA3_384,
+  kSHA3_512,
+  kSHAKE128,
+  kSHAKE256,
+  kcSHAKE128,
+  kcSHAKE256
+} kmac_operation_t;
+
+// This struct is used to pack NIST test vector neatly
+// This probably needs to be defined in another header later
+typedef struct kmac_test_vector {
+  kmac_operation_t operation;
+  char func_name[36];
+  char custom_str[36];
+  uint8_t *key;
+  size_t key_len;
+  uint8_t *input_str;
+  size_t input_str_len;
+  uint8_t *digest;
+  size_t digest_len;
+} kmac_test_vector_t;
+
+// The fields of this enum needs to be revised later (not all errors cases are used)
+typedef enum kmac_error {
+  kKmacOk = 0,
+  kKmacInternalError = 1,
+  kKmacArgsError = 2,
+  kKmacNotImplemented = 3,
+} kmac_error_t;
+
+/*
+* Sets the CFG register of KMAC for given `operation_type`. The struct type kmac_operation_t is defined in a way that the key size of the operation is also part of it. 
+* For instance, if we want to run SHA-3 with 224-bit digest size, then `operation_type` = kSHA3_224.
+* 
+* @param operation_type The chosen operation, see kmac_operation_t struct.
+* @return Error of type kmac_error_t.
+*/
+OT_WARN_UNUSED_RESULT
+kmac_error_t kmac_init(kmac_operation_t operation_type);
```

Comment:
No action required, but just a thought here: I think there's a fairly big design choice to be made here with how to represent different operations. Namely, is it better to use the same `init` and `update` functions for all operation types (e.g. `kmac_init(kSha3384)`, or have separate functions for each (e.g. `kmac_sha3_384_init()`), or something in between (e.g. `kmac_sha3_init(kSha3Length384)`)? There's no right or wrong answer to that, but if you find yourself having a lot of switch statements it might be a change worth considering.

Reply:
This is not final and likely to change as I add cSHAKE and KMAC. For the moment, I suggest to add a comment on top of the `kmac_operation_t`. I think we can make better intermediate-API choices once we try to bridge between this driver and the higher level crypto API. 
Also, to me it seems that we either put this switch-case statements inside the `init` and keep it as it is, or we defer the switch-case part to higher level API implementation. Therefore, I don't see the benefit of trying to avoiding switch-case here.